ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx

```diff
@@ -10,66 +11,71 @@ require('./workflows-toolbar.scss');
-                        <div className='workflows-toolbar__count'>
-                            {this.getNumberSelected() === 0 ? 'No' : this.getNumberSelected()}
-                            &nbsp;workflow{this.getNumberSelected() === 1 ? '' : 's'} selected
-                        </div>
-                        <div className='workflows-toolbar__actions'>{this.renderActions(ctx)}</div>
-                    </div>
-                )}
-            </Consumer>
-        );
-    }
+    const groupAction = useMemo<WorkflowsOperation[]>(() => {
+        const actions: any = Actions.WorkflowOperationsMap;
+        const disabled = props.actionsIsDisabled;
 
-    private getNumberSelected(): number {
-        return this.props.selectedWorkflows.size;
-    }
+        return Object.keys(actions).map((actionName: WorkflowOperationName) => {
+            const action = actions[actionName];
+            return {
+                title: action.title,
+                iconClassName: action.iconClassName,
+                isDisabled: disabled[actionName],
+                action,
+                workflowsAction: async () => {
+                    // check for action
+                    const confirmed = await popup.confirm('Confirm', `Are you sure you want to ${action.title.toLowerCase()} all selected workflows?`);
+                    if (!confirmed) {
+                        return;
+                    }
 
-    private async performActionOnSelectedWorkflows(ctx: ContextApis, title: string, action: WorkflowOperationAction): Promise<any> {
-        const confirmed = await ctx.popup.confirm('Confirm', `Are you sure you want to ${title.toLowerCase()} all selected workflows?`);
-        if (!confirmed) {
-            return Promise.resolve(false);
-        }
+                    // check for delete from archived workflows
+                    let deleteArchived = false;
+                    if (action.title === 'DELETE') {
+                        // check for delete workflows from archived workflows
+                        for (const entry of props.selectedWorkflows) {
+                            if (isArchivedWorkflow(entry[1])) {
+                                deleteArchived = await popup.confirm('Confirm', 'Do you also want to delete them from the Archived Workflows database?');
+                                break;
+                            }
+                        }
+                    }
 
-        let deleteArchived = false;
-        if (title === 'DELETE') {
-            for (const entry of this.props.selectedWorkflows) {
-                if (isArchivedWorkflow(entry[1])) {
-                    deleteArchived = await ctx.popup.confirm('Confirm', 'Do you also want to delete them from the Archived Workflows database?');
-                    break;
-                }
-            }
-        }
+                    performActionOnSelectedWorkflows(action.title, action.action, deleteArchived);
```

Comment:
```suggestion
                    await performActionOnSelectedWorkflows(action.title, action.action, deleteArchived);
```

the previous behavior waited for this to complete first